src/librustc_codegen_ssa/lib.rs

```diff
@@ -10,34 +10,19 @@
 #![feature(nll)]
 #![allow(unused_attributes)]
 #![allow(dead_code)]
+#![deny(rust_2018_idioms)]
+#![allow(explicit_outlives_requirements)]
+#![allow(elided_lifetimes_in_paths)]
```

Comment:
This crate uses `TyCtxt` more than a few times, so `elided_lifetimes_in_paths` is allowed.